Userland/Libraries/LibWeb/WebAssembly/WebAssemblyObject.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Runtime/Object.h>
+#include <LibWasm/AbstractMachine/AbstractMachine.h>
+#include <LibWeb/Forward.h>
+
+namespace Web::Bindings {
+
+class WebAssemblyObject final : public JS::Object {
+    JS_OBJECT(WebAssemblyObject, JS::Object);
+
+public:
+    explicit WebAssemblyObject(JS::GlobalObject&);
+    virtual void initialize(JS::GlobalObject&) override;
+    virtual ~WebAssemblyObject() override = default;
+
+    struct CompiledWebAssemblyModule {
+        CompiledWebAssemblyModule(ByteBuffer&& source, Wasm::Module&& module)
+            : source(move(source))
+            , module(move(module))
+        {
+        }
+
+        ByteBuffer source; // This is kept around because the spec wants it at [[Source]]
```

Comment:
the spec doesn't care what we do with internal properties though, if this isn't used anywhere there's no need to keep it around.